flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/exceptions.scala

```diff
@@ -165,3 +165,31 @@ case class AmbiguousTableSourceConverterException(
 
   def this(tableType: String) = this(tableType, null)
 }
+
+/**
+  * Exception for operation on a nonexistent external catalog
+  *
+  * @param catalogName external catalog name
+  * @param cause
+  */
+case class ExternalCatalogNotExistException(
```

Comment:
hi, @twalthr , do you means that we could introduce two superclass, which is AlreadyExistException and NotExistException, other exceptions are subclass of these two exception class? IMO, there is no need to introduce a superclass, because each exception represents a different unexpected situation that user may cares, they may want to know whether database or table or something else does not exist yet. what's your opinion?